### PR TITLE
[#48604] Add "I am member" filter for projects

### PR DESCRIPTION
**Pull request by aaron-contreras**
**NOTE FOR REVIEWER**: This PR builds on the work done in https://github.com/opf/openproject/pull/13073. Therefore, only the [last commit](bb6a148fb22c051d004aa6e6ecad8a3273682d57) is part of this PR (at least until https://github.com/opf/openproject/pull/13073 is merged into dev)

See Work Package: https://community.openproject.org/work_packages/48604

### CODE REVIEW

**Comment by aaron-contreras** [comment]
First time creating a filter. Looking for comments/suggestions dropped here on both the filter's implementation/spec 🙏

File: app/models/project.rb [hunk 1]
```diff
@@ -157,6 +157,8 @@ class Project < ApplicationRecord
   scope :visible, ->(user = User.current) { where(id: Project.visible_by(user)) }
   scope :newest, -> { order(created_at: :desc) }
   scope :active, -> { where(active: true) }
+  scope :with_member, ->(user = User.current) { where(id: user.memberships.pluck(:project_id)) }
+  scope :without_member, ->(user = User.current) { where.not(id: user.memberships.pluck(:project_id)) }
```
Comment (ulferts) [inline app/models/project.rb]: Using `pluck` in this context leads to an unnecessary extra request as the `pluck` has to be resolved first with the results of that then being passed into the next query. Using `select` avoids that.

**Review by ulferts — CHANGES_REQUESTED** [review]
Only a tiny thing @aaron-contreras. Merge at will after addressing it.

**Review by ulferts — APPROVED** [review]
👍

**Pull request merged**